static/app/components/forms/fields/choiceMapperField.tsx

```diff
@@ -166,30 +171,31 @@ export default class ChoiceMapperField extends Component<ChoiceMapperFieldProps>
       addDropdown.items?.filter(i => !value.hasOwnProperty(i.value)) ?? [];
 
     const valueMap =
-      addDropdown.items?.reduce((map, item) => {
+      addDropdown.items?.reduce<Record<string, React.ReactNode>>((map, item) => {
         map[item.value] = item.label;
         return map;
       }, {}) ?? {};
 
     const dropdown = (
-      <DropdownAutoComplete
+      <CompactSelect
         {...addDropdown}
-        alignMenu={valueIsEmpty ? 'right' : 'left'}
-        items={selectableValues}
-        onSelect={addRow}
-        disabled={disabled}
-      >
-        {({isOpen}) => (
-          <DropdownButton
-            icon={<IconAdd isCircled />}
-            isOpen={isOpen}
-            size="xs"
-            disabled={disabled}
-          >
-            {addButtonText}
-          </DropdownButton>
-        )}
-      </DropdownAutoComplete>
+        emptyMessage={
+          selectableValues.length === 0
+            ? addDropdown.emptyMessage
+            : addDropdown.noResultsMessage
+        }
+        size="xs"
+        searchable
+        disabled={false}
```

Comment:
note: we have to pass `disabled={false}` because per default, the `CompactSelect` disables itself if it has no options. However, we reduce options here after selection and we still want the field to remain interactive